array_methods_assignmen.js

```diff
@@ -0,0 +1,404 @@
+//
+// ```
+// Input: ["a", 123, "b", "c", {name: "cat"}]
+// Output: ["a", "", "b", "c", ""]
+// ```
+//
+// let arr2 = ["a", 123, "b", "c", {name: "cat"}];
+// let allStringArr = arr2.map((crrElem, index) => {
+// if (typeof crrElem === 'number' || typeof crrElem === 'object' ) {
+//  return " "
+// } else {
+//  return  crrElem
+// }
+// })
+// console.log(allStringArr)
+
+
+// 3. Write a function that returns only the even elements in an array of numbers.
+//
+// ```
+// Input: [1,2,3,4,5]
+// Output: [2,4]
+//  let arr3 = [1,2,3,4,5];
+// let filteredArr3 = arr3.filter((currentElement3) => {
+//  if (currentElement3 % 2 ===0) {
+//     return true
+//    } else {
+//     return false
+//     }
+//    })
+//  console.log(filteredArr3)
+
+//===============
+// let array3 =  [1,2,3,4,5]
+// let evenNums = array3.filter((num, index) => {
+//  return num % 2 === 0
+// })
+
+// 4. Write a function that checks if all
+//elements in the array are the same.
+//
+// ```
+// Input: [1,1,1,1,1]
+// Output: true
+// let arr4 = [1,1,1,1,1];
+// let arr4same = arr4.every(elem => {
+//   if (elem === 1 ){
+//     return true
+//   }
+//   return false
+// })
+// console.log(arr4same)
+// //
+// Input: [1,1,1,1,2]
+// Output: false
+// ```
+// let arr5 = [1,1,1,1,2]
+// let arr5same = arr5.every(elem => {
+//   if (elem === 1 ){
+//     return true
+//   }
+//   return false
+// })
+// console.log(arr5same)
+// 5. Write a function that returns the sum of
+//every element in an array.
+//
+// ```
+// Input: [1,2,3,4,5]
+// Output: 15
+//function adder (total,crrElem){
+//   return total + currElem;
+// }
+// let mums =[ 1, 2, 3, 4, 5];
+// let largest = nums.reduce((acc,crrElem)=>{
+//   console.log(`acc: ${acc}, crr: ${crrElem}`)
+//   if (crrElem > acc) {
+//     return crrElem
+//   } else{
+//     return acc;
+//   }
+// })
+// console.log(largest);
+// ```
+//
+// 6. Write a function that finds largest number in an array.
+//
+// ```
+// Input: [1,2,6,3,4,5]
+// Output: [6]
+//
+//const arr6 = [1,2,6,3,4,5];
+// const largest = arr6.reduce((prev, curr) => {
+// return Math.max(prev, curr)
+// });
+// console.log(largest);
+//=====================> need to ask JR
+// let arr6 = [1,2,6,3,4,5];
+// //let result6 = largestElm(arr6);
+// let largest = arr6.reduce((largestElm, currElm) =>{
+//   if (largestElm > currElm){
+//     return largestElm;
+//   } else {
+//     return currElm
+//   }
+// });
+// console.log(largest);
+//========================================
+// 7. Write a function `countZeroes`, which takes an array
+// of numbers as its argument and returns the amount of zeroes
+//that occur in it.
+// ```
+// Input: [1,0,0,2,0,3,0,0,0]
+// Output: 6
+// ```
+// let array7 = [1,0,0,2,0,3,0,0,0];
+//
+// const countZeroes = (array, num) => {
+// let count = array.filter(element => element === num)
+// return count.length;
+// }
+//
+// let result7 = countZeroes(array7, 0);
+// console.log(result7);
+// ==============================
```

Comment:
Very close. Did you know you can give an argument a default value? Your code will work if you give the num argument a default value of 0. Because you are comparing num(0) to each element of the array